src/views/Order.vue

```diff
@@ -154,62 +155,71 @@
+                        <ion-label>{{ $t("Purchase order") }}</ion-label>
                         <ion-chip slot="end">
+                          <!-- TODO: check for this property -->
                           <ion-label>PO#</ion-label>
                         </ion-chip>
                       </ion-item>
-                      <ion-item>
-                        <ion-label> {{ $t("Estimated arrival") }} </ion-label>
-                        <p slot="end">14 Jan 2021</p>
+                      <ion-item v-if="item.orderItemStatusId !== 'ORDER_COMPLETED'">
+                        <ion-label>{{ $t("Estimated arrival") }}</ion-label>
+                        <!-- TODO: handle it property again -->
+                        <p slot="end">{{ item.promiseOrderArrivalDate ? $filters.formatUtcDate(item.promiseOrderArrivalDate, 'YYYY-MM-DDTHH:mm:ssZ', 'D MMM YYYY') : '-' }}</p>
                       </ion-item>
-                      <ion-item>
+                      <ion-item v-if="item.orderItemStatusId !== 'ORDER_COMPLETED'">
                         <ion-label> {{ $t("Promise date") }} </ion-label>
-                        <p slot="end">14 Jan 2021</p>
+                        <p slot="end">{{ item.promisedDatetime ? $filters.formatUtcDate(item.promisedDatetime, 'YYYY-MM-DDTHH:mm:ssZ', 'D MMM YYYY') : '-'  }}</p>
                       </ion-item>
-                      <ion-item>
+                      <ion-item v-if="item.orderItemStatusId !== 'ORDER_COMPLETED'">
                         <ion-label> {{ $t("Auto cancel") }} </ion-label>
-                        <p slot="end">14 Jan 2021</p>
+                        <p slot="end">{{ item.autoCancelDate ? $filters.formatUtcDate(item.autoCancelDate, 'YYYY-MM-DDTHH:mm:ssZ', 'D MMM YYYY') : "-" }}</p>
+                      </ion-item>
+                      <ion-item v-if="item.orderItemStatusId === 'ORDER_COMPLETED'">
+                        <ion-label> {{ $t("Received Date") }} </ion-label>
+                        <p slot="end">{{ item.receivedDate ? $filters.formatUtcDate(item.receivedDate, 'YYYY-MM-DDTHH:mm:ssZ', 'D MMM YYYY') : "-" }}</p>
                       </ion-item>
-                      <ion-buttons>
+                      <!-- TODO: make edit date button functional, also add UI for same -->
+                      <!-- <ion-buttons>
                         <ion-button color="primary" fill="clear">{{ $t("Edit dates") }}</ion-button>
-                      </ion-buttons>
+                      </ion-buttons> -->
                     </ion-list>
                   </ion-card>
                   <ion-card>
                     <ion-list>
                       <ion-list-header>{{ $t("Fulfillment") }}</ion-list-header>
                       <ion-item>
                         <ion-label> {{ $t("Shipping method") }} </ion-label>
-                        <p>Standard</p>
+                        <p>{{ item.shipmentMethodTypeId ? item.shipmentMethodTypeId : "-"}}</p>
```

Comment:
This should be the shipment method description
We need to gather this info from ShipmentMethodType entity

Reply:
Sir, created a state for storing the shipment methods and use that on details page